### ISSUE TITLE
pll_translate_string() by name, not string

### ISSUE THREAD

**Issue by jrtderonde** [polylang/polylang#375]
## Prerequisites

- [x] This is not a usage question (Those should be directed to the [community supported forum](https://wordpress.org/support/plugin/polylang), unless this is a question about Polylang Pro in which case you should [use the helpdesk](https://polylang.pro/support/)).
- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x] The issue still exists against the latest `master` branch of Polylang and the latest WordPress version.
- [ ] This bug happens with only Polylang plugin active
- [ ] This bug happens with a default WordPress theme active
- [ ] I can reproduce this bug consistently

Basically not an issue, more somewhat of a request. `pll_translate_string()` uses the `$string` instead of the `$name` which is pretty strange. For instance; when a string is parsed in the theme as `pll_translate_string('Hello World', 'en')` it will not return the translation anymore whenever the original string is i.e. changed to `'Hello Planet'`.

I'm requesting either a change or a new function in order to translate by registered `$name`... I don't really see the point of registering a `$name` if it's not possible to translate it by this.

**Comment by Chouby** [comment]
The translation functions, `pll__()`, `pll_translate_string()` are based on the WP function `__()`  which translates the string, given only that string. When you read a .po file, there is nothing more than the string and the translation. If you change the string in the code, then you need to update the translation too. You should think Polylang strings translations as a .po file.

In Polylang, the name and group are used only in the translations list table to improve sorting ans search. They are not used in the translation process. 

Your request makes me think that you are used to WPML and are looking to make Polylang work like WPML. Then just use the WPML API, it should work as you expected. See: https://github.com/polylang/polylang/blob/2.6.1/modules/wpml/wpml-legacy-api.php#L270-L289

**Comment by edwinhuish** [comment]
@Chouby It's no need to be same as WPML, but it's very important to add ```$group``` or ```$domain``` to be the second parameter as ``` __( string $text, string $domain = 'default' ) ```. Because the translation will be totally different in different situation.

Also, I'm agree with @jrtderonde , Polylang should be like WPML function, no WP function ```__()```, because .po file normally use for display text. But WPML or Polylang usually use for admin plugin setting fields and display text. For the setting fields, it's very easy changed it accidentally. If happen, even just a point or space, we need to translate the hold translation again... If the translation is long long story...... 

By the another side, I think it's slowly comparing long strings than a short string name.

**Comment by kresnasatya** [comment]
Hi, I have the same thought with the author of this issue. Although I see this issue can't be fix, how about if Polylang add another function like `pll_translate($name)` which accept `$name` param that comes from `pll_register_string()`?

**Comment by karamgaby** [comment]
I agree, text translation depends on context because translations would be different by context

there should be a way to provide context to the translator 

_x( string $text, string $context, string $domain = ‘default’ )
